### PR TITLE
Improve inputs for headers, query params, to stretch when focused (like HTTPie)

### PR DESCRIPTION
**Pull request by brettimus**
## TODO

- [x] Press `ESC` to blur input
- [x] Investigate why newlines are possible in FireFox
- [x] Test safari.... 🙈 

## Summary

This PR introduces a new input-like component that we can use for adding query/path parameters and headers in their respective forms.

It also factors out the codemirror editor components into their own module in the `components` directory

https://github.com/user-attachments/assets/63fa1608-2458-43bb-8e2d-255efe4a6aa2

### CODE REVIEW

**Comment by brettimus** [comment]
image

this is the blocker before merging. something is weird with codemirror and breaking the layout every once in a while

**Comment by brettimus** [comment]
another issue: tab inserts a tab instead of going to next component

**Comment by brettimus** [comment]
Hmmm one annoying thing that came up is that the codemirror inputs swallow our keyboard shortcuts (cmd+g and cmd+b) when they have focus

**Review by mies — APPROVED** [review]

**Review by laulauland — APPROVED** [review]
Looks good! Ideally we'd "blur" the text input when you hit Escape so that it would unblock using the cmd+b and other keyboard shortcuts.

**Comment by brettimus** [comment]
> Looks good! Ideally we'd "blur" the text input when you hit Escape so that it would unblock using the cmd+b and other keyboard shortcuts.

did it!

File: studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx
Comment (flenter) [inline studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx]: Curious, can this parameter be removed?

File: studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx
Comment (flenter) [inline studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx]: Bonus points if the static styling is done via the tailwind classes (which is also done)  and even more points if isFocused can be removed (something like:`focus-within:overlfow-auto`)

**Review by flenter — APPROVED** [review]
nice!

File: studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx
Comment (brettimus) [reply studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx]: good point — this style object was a holdover from the first attempts at writing a custom component. we can just set the width with a tailwind class that's passed as a prop! refactoring

Comment (brettimus) [reply studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx]: focus might be harder though, since the parent div needs to be aware of focus state, but i will try

Comment (brettimus) [reply studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx]: yeah so focus state needed to remain in order to conditionally load a codemirror extension that helps with truncation. but the styles have been refactored to rely on focus-within!

File: studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx
Comment (brettimus) [reply studio/src/components/CodeMirrorEditor/CodeMirrorInput.tsx]: indeed it can - was toying with it to see if i had additional info on (e.g.) whether or not "mod" was pressed as well

**Pull request merged**
